dali/test/python/test_operator_decoders_image.py

```diff
@@ -90,26 +88,124 @@ def log(img_type, size, device, threads):
+                               output_type = types.RGB)
+    # ToDo make window random?
+    images_crop_2 = fn.crop(images,
+                            crop = (224, 224),
+                            crop_pos_x = 0.3,
+                            crop_pos_y = 0.2)
+
+    return images_crop_1, images_crop_2
+
+@pipeline_def
+def create_decoder_random_crop_pipeline(data_path, device):
+    seed = 1234
+    jpegs, _ = fn.readers.file(file_root = data_path,
+                               shard_id = 0,
+                               num_shards = 1,
+                               name = "Reader")
+
+    images_random_crop_1 = fn.decoders.image_random_crop(jpegs,
+                                                         device = device,
+                                                         output_type = types.RGB,
+                                                         hw_decoder_load = 0.7,
+                                                         seed = seed)
+    images_random_crop_1 = fn.resize(images_random_crop_1, size = (224,224))
+
+    images = fn.decoders.image(jpegs,
+                               device = device,
+                               hw_decoder_load = 0.7,
+                               output_type = types.RGB)
+    images_random_crop_2 = fn.random_resized_crop(images, size = (224,224), seed = seed)
+
+    return images_random_crop_1, images_random_crop_2
+
+def run_decode_fused(test_fun, path, img_type, batch, device, threads, validation_fun):
+    data_path = os.path.join(test_data_root, path, img_type)
+    pipe = test_fun(data_path=data_path, batch_size=batch, num_threads=threads, device_id=0, device=device, prefetch_queue_depth=1)
+    pipe.build()
+    iters = math.ceil(pipe.epoch_size("Reader") / batch)
+    for _ in range(iters):
+        out_1, out_2 = pipe.run()
+        for img_1, img_2 in zip(out_1, out_2):
+            img_1 = to_array(img_1)
+            img_2 = to_array(img_2)
+            # random_resized_crop can properly handle border as it has pixels that are cropped out, while
+            # plain resize folowing image_decoder_random_crop cannot do that and must duplicate the border pixels
+            #assert np.mean(np.abs(img_1 - img_2) < 0.5)
+            assert validation_fun(img_1, img_2)
+
+def test_image_decoder_fused():
+    for test_fun in [create_decoder_slice_pipeline, create_decoder_crop_pipeline, create_decoder_random_crop_pipeline]:
+        if test_fun == create_decoder_random_crop_pipeline:
+            # random_resized_crop can properly handle border as it has pixels that are cropped out, while
+            # plain resize folowing image_decoder_random_crop cannot do that and must duplicate the border pixels
+            validation_fun = lambda x, y: np.mean(np.abs(x - y) < 0.5)
+        else:
+            validation_fun = lambda x, y: np.allclose(x, y)
+        for device in {'cpu', 'mixed'}:
+            threads = 4
+            batch_size = 10
```

Comment:
```suggestion
            threads = 4
            batch_size = 10
```
nitpick: Those declarations could be done outside of the loop

Reply:
Done